v0/destinations/hs/transform.js

```diff
@@ -91,30 +91,46 @@ const processRouterDest = async inputs => {
     const respEvents = getErrorRespEvents(null, 400, "Invalid event array");
     return [respEvents];
   }
+
   const successRespList = [];
   const errorRespList = [];
   // using the first destination config for transforming the batch
   const { destination } = inputs[0];
   let propertyMap;
   const mappedToDestination = get(inputs[0].message, MappedToDestinationKey);
-  const { objectType } = getDestinationExternalIDInfoForRetl(inputs[0].message, "HS");
-  if (
-    mappedToDestination &&
-    GENERIC_TRUE_VALUES.includes(mappedToDestination?.toString())
-  ) {
-    // skip splitting the batches to inserts and updates if object it is an association
-    if (objectType.toLowerCase() !== "association") {
-      // get info about existing objects and splitting accordingly.
-      inputs = await splitEventsForCreateUpdate(inputs, destination);
-    }
-  } else {
-    // reduce the no. of calls for properties endpoint
-    const traitsFound = inputs.some(input => {
-      return fetchFinalSetOfTraits(input.message) !== undefined;
-    });
-    if (traitsFound) {
-      propertyMap = await getProperties(destination);
+  const { objectType } = getDestinationExternalIDInfoForRetl(
```

Comment:
Do we need to put this function in try-catch ?

Reply:
No, we are just getting data from payload there is no axios call in it

Reply:
We'll have to be careful even of the unhandled error that could happen without axios call present in the code. Particularly in router transformation destinations, as there is no outer error catch block to safeguard against such failures.

Reply:
For now, I will resolve this comment but we need to keep in mind about this while writing code for future destinations